### PR TITLE
Catalyst support

### PR DESCRIPTION
**Pull request by ChristosT**
## Summary
This PR completes the work of @andrewcombs and @c-wetterer-nelson from #3444 by adding a docker image that includes ParaView and can be used for testing.

It also adds a minor fix a02c1a5110ce6104a16aba4eee4685c9196eb758 discovered when testing these changes with the current master of WarpX.

## Additional background
Supersedes and closes #3444 

## Checklist

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Review by ax3l — APPROVED** [review]
That looks great, thank you!

Sorry for the slow reply due to vacation and work travel in the last month.

**Comment by ax3l** [comment]
Hi @ChristosT, do you plan to also replace https://github.com/ECP-WarpX/WarpX/pull/4121 with a new PR? :star_struck: 
It sounds like you already got WarpX running with Catalyst? :star_struck:

File: .github/workflows/catalyst.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  catalyst:
+    name: Catalyst
+    runs-on: ubuntu-22.04
+    if: github.event.pull_request.draft == false
+    env:
+      CXX: g++
+      CC: gcc
+      CMAKE_PREFIX: "/opt/conduit;/opt/catalyst"
```
Comment (ax3l) [inline .github/workflows/catalyst.yml:17]: @ChristosT did you mean to write here
```suggestion
      CMAKE_PREFIX_PATH: "/opt/conduit:/opt/catalyst"
```
?

Comment (ax3l) [reply .github/workflows/catalyst.yml:17]: Posting this to #4067

**Comment by ChristosT** [comment]
> Hi @ChristosT, do you plan to also replace https://github.com/ECP-WarpX/WarpX/pull/4121 with a new PR? :star_struck: 
> It sounds like you already got WarpX running with Catalyst? :star_struck: 

Yes ! I have it running . But I am on vacation right now. I will come back to it before the end of the month :)

**Pull request merged**